### ISSUE TITLE
You code is outdated!

### ISSUE THREAD

**Issue by korolaab** [philipperemy/keras-attention-mechanism#22]
Your code doesn't fit to new versions of keras
To fix it change those strings in "attention_dense.py":

1. **"from keras.layers import Input, Dense, merge"** on **"from keras.layers import Input, Dense,multiply"**;
2. **"attention_mul = merge([inputs, attention_probs], output_shape=32, name='attention_mul', mode='mul')"** on **"attention_mul = multiply([inputs, attention_probs],name='attention_mul')" ;
and in "attention_lstm.py"**:
in"attention_lstm.py":

1. 1. **import multiply too**;
2. 2. **"output_attention_mul = merge([inputs, a_probs], name='attention_mul', mode='mul')"** change on **"output_attention_mul = multiply([inputs, a_probs], name='attention_mul')"**

**Comment by YanShulinjj** [comment]
what version of keras should replace these points?

**Comment by JunkyByte** [comment]
> This is not replacing the original purpose of the code I think. Instead of using `multiply`, you should using `Concatenate` or `Add`.

If you look at the `mode` params of the merge call it is `mul` which makes multiplication.
So the answer from @korolaab is correct (also if you look at the paper / implementations it is pretty clear that there's a multiplication there as you combine attention and input, attention is a prob. distribution (as is the output of a `softmax` activation).

**Comment by shangbecky** [comment]
I agree with @korolaab  and @JunkyByte

**Comment by philipperemy** [comment]
Code has been updated. Thanks guys :)
